src/api/client.js

```diff
@@ -200,6 +277,68 @@ export class Client {
         schema: Schemas.REPO_ARRAY,
         paginationArgs: [userId],
       }),
+
+    starRepo: (repoId: string, params: SpecialParameters = {}) =>
+      this.put({
+        endpoint: `user/starred/${repoId}`,
+        params,
+        schema: Schemas.GQL_REPO,
+        updateEntity: {
+          type: 'gqlRepos',
+          id: repoId,
+          updater: gqlRepo => ({
+            isStarred: true,
+            stargazersCount: gqlRepo.stargazersCount + 1,
+          }),
+        },
+      }),
+    unstarRepo: (repoId: string, params: SpecialParameters = {}) =>
+      this.delete({
+        endpoint: `user/starred/${repoId}`,
+        params,
+        schema: Schemas.GQL_REPO,
+        updateEntity: {
+          type: 'gqlRepos',
+          id: repoId,
+          updater: gqlRepo => ({
+            isStarred: false,
+            stargazersCount: gqlRepo.stargazersCount - 1,
+          }),
+        },
+      }),
+    watchRepo: (repoId: string, params: SpecialParameters = {}) =>
+      this.put({
+        endpoint: `repos/${repoId}/subscription`,
+        params,
+        fetchParameters: {
+          body: {
+            subscribed: true,
+          },
+        },
+        schema: Schemas.GQL_REPO,
+        updateEntity: {
+          type: 'gqlRepos',
+          id: repoId,
+          updater: gqlRepo => ({
+            isSubscribed: true,
+            watchersCount: gqlRepo.watchersCount + 1,
```

Comment:
Good stuff :)

Reply:
this was almost already in #770, but changed it to take a callback with the modified elements, for way more flexibility.